### ISSUE TITLE
Node.js API should allow fixing a list of files in-place

### ISSUE THREAD

**Issue by jeff-held-aiven** [stylelint/stylelint#7684]
### What is the problem you're trying to solve?

I'm trying to use Stylelint's Node.js API to format my SCSS files programmatically. I can see that I can specify the `fix` option when calling `lint`, but this seems to be useful only if you pass `code` to `lint` ([as shown in the docs](https://stylelint.io/user-guide/node-api#example-f)). What I instead want to do is the following:

```js
import Stylelint from 'stylelint';

await Stylelint.lint({
  files: 'src/**/*.scss',
  fix: true,
});
// Nothing else to be done; all files should be formatted and overwritten with the fixes
```

### What solution would you like to see?

I would like to be able to do what I described, [similar to what ESLint allows](https://eslint.org/docs/latest/integrate/nodejs-api#:~:text=Here%E2%80%99s%20an%20example%20that%20autofixes%20lint%20problems).

**Comment by ybiquitous** [comment]
@jeff-held-aiven The combination of `files` and `fix` works now, doesn't it?

**Comment by ybiquitous** [comment]
### Demo

`test.mjs`:

```js
import stylelint from 'stylelint';

await stylelint.lint({
	files: ['test.css'],
	config: { rules: { 'hue-degree-notation': 'angle' } },
	fix: true,
});
```

Run:

```sh-session
$ cat test.css
a {
  color: hsl(198 28% 50%);
}

$ node test.mjs

$ cat test.css
a {
  color: hsl(198deg 28% 50%);
}

$ npx stylelint -v
16.5.0
```

Diff:

```diff
--- a/test.css
+++ b/test.css
@@ -1,3 +1,3 @@
 a {
-  color: hsl(198 28% 50%);
+  color: hsl(198deg 28% 50%);
 }
```
